File: react-native/client/ios/Records/AccountRecord.swift
import CoinbaseWalletSDK
import ExpoModulesCore
import Foundation

struct AccountRecord : Record {
    @Field
    var chain: String

    @Field
    var networkId: Int

    @Field
    var address: String
}

extension Account {
    var asRecord: AccountRecord.Dict {
        let record = AccountRecord()
        record.chain = self.chain
        record.networkId = Int(self.networkId)
        record.address = self.address

        return record.toDictionary()
    }
}

File: react-native/client/android/src/main/java/expo/modules/coinbasewalletsdkexpo/records/AccountRecord.kt
package expo.modules.coinbasewalletsdkexpo.records

import com.coinbase.android.nativesdk.message.request.Account
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class AccountRecord : Record {
    @Field
    var chain: String = ""

    @Field
    var networkId: Int = 1

    @Field
    var address: String = ""
}

val AccountRecord.asAccount: Account
    get() {
        return Account(
            chain = this.chain,
            networkId = this.networkId.toLong(),
            address = this.address
        )
    }

val Account.asRecord: AccountRecord
    get() {
        val record = AccountRecord()
        record.chain = chain
        record.networkId = networkId.toInt()
        record.address = address
        return record
    }
